units/task_monitor_execution_timeout.go

```diff
@@ -0,0 +1,152 @@
+package units
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/evergreen-ci/evergreen"
+	"github.com/evergreen-ci/evergreen/apimodels"
+	"github.com/evergreen-ci/evergreen/model"
+	"github.com/evergreen-ci/evergreen/model/host"
+	"github.com/evergreen-ci/evergreen/model/task"
+	"github.com/mongodb/amboy"
+	"github.com/mongodb/amboy/dependency"
+	"github.com/mongodb/amboy/job"
+	"github.com/mongodb/amboy/registry"
+	"github.com/mongodb/grip"
+	"github.com/mongodb/grip/message"
+	"github.com/mongodb/grip/sometimes"
+	"github.com/pkg/errors"
+)
+
+const (
+	heartbeatTimeoutThreshold   = 7 * time.Minute
+	taskExecutionTimeoutJobName = "task-execution-timeout"
+)
+
+func init() {
+	registry.AddJobType(taskExecutionTimeoutJobName, func() amboy.Job {
+		return makeTaskExecutionTimeoutMonitorJob()
+	})
+}
+
+type taskExecutionTimeoutJob struct {
+	job.Base `bson:"metadata" json:"metadata" yaml:"metadata"`
+}
+
+func makeTaskExecutionTimeoutMonitorJob() *taskExecutionTimeoutJob {
+	j := &taskExecutionTimeoutJob{
+		Base: job.Base{
+			JobType: amboy.JobType{
+				Name:    taskExecutionTimeoutJobName,
+				Version: 0,
+			},
+		},
+	}
+
+	j.SetDependency(dependency.NewAlways())
+	return j
+}
+
+func NewTaskExecutionMonitorJob(id string) amboy.Job {
+	j := makeTaskExecutionTimeoutMonitorJob()
+	j.SetID(fmt.Sprintf("%s.%s", taskExecutionTimeoutJobName, id))
+	return j
+}
+
+func (j *taskExecutionTimeoutJob) Run(ctx context.Context) {
+	defer j.MarkComplete()
+
+	flags, err := evergreen.GetServiceFlags()
+	if err != nil {
+		j.AddError(err)
+		return
+	}
+
+	if flags.MonitorDisabled {
+		grip.InfoWhen(sometimes.Percent(evergreen.DegradedLoggingPercent), message.Fields{
+			"message": "monitor is disabled",
+			"impact":  "skipping task heartbeat cleanup job",
+			"mode":    "degraded",
+		})
+		return
+	}
+
+	tasks, err := task.Find(task.ByStaleRunningTask(heartbeatTimeoutThreshold))
+	if err != nil {
+		j.AddError(errors.Wrap(err, "error finding tasks with timed-out or stale heartbeats"))
+		return
+	}
+
+	for _, task := range tasks {
+		msg := message.Fields{
+			"operation": j.Type().Name,
+			"runner":    "monitor",
```

Comment:
Does it make sense to continue logging that this is the monitor? I think we should log by job type